modules/core/src/main/scala/gem/config/DynamicConfig.scala

```diff
@@ -7,22 +7,22 @@ import java.time.Duration
 
 sealed trait SmartGcalKey
 
-sealed abstract class InstrumentConfig extends Product with Serializable {
+sealed abstract class DynamicConfig extends Product with Serializable {
+
+  type I <: Instrument with Singleton
```

Comment:
It's a marker trait that the compiler knows, for singleton types. So this means `I` can only be a type like `Instrument.GmosSouth.type` (or a bogus subtype like `... with SomethingElse` or `Nothing` but these types are uninhabited). It can't be `Instrument` for instance. I don't know that we really need it here, it's mostly documentation that you should expect `I` to be a singleton type.